tasks/human_eval_056.rs

```diff
@@ -9,7 +9,53 @@ use vstd::prelude::*;
 
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+spec fn spec_bracketing_helper(brackets: Seq<char>) -> (int, bool) {
+    brackets.fold_left((0, true), |p: (int, bool), c| {
+        let (x, b) = p;
+        match (c) {
+            '<' => (x + 1, b),
+            '>' => (x - 1 , b && x == 0),
+            _ => (x, b),
+        }
+    })
+}
+
+spec fn spec_bracketing(brackets: Seq<char>) -> bool {
+    let p = spec_bracketing_helper(brackets);
+    p.1 && p.0 == 0
+}
+
+fn correct_bracketing(brackets: &str) -> (ret: bool)
+    requires
+        brackets@.len() <= i32::MAX
+        -brackets@.len() >= i32::MIN
```

Comment:
Its to prevent overflow of the stack size local variable.
`-brackets@.len() >= i32::MIN` isn't strictly necessary, but in my case stack_size keeps decrementing if there are lots of `>`. I thought its nicer to keep it symmetric.